examples/contour/02-role-contour.yaml

```diff
@@ -13,6 +13,7 @@ rules:
   - ""
   resources:
   - configmaps
+  - events
```

Comment:
Is this for the controller-runtime manager to write events?

Reply:
yep, without it, leader election still works but theres errors in the logs from failures to write events

Reply:
Might be good to add this to a role instead of a cluster role, I think for this specifically, it doesn't need cluster-wide access.

Reply:
hm yeah thats a good point, might open a separate PR to address this since all of the RBAC for leaderelection resources could be namespace scoped, but it makes the role generation a little different

Reply:
heres what things might look like if we switched to more scoped resources for the leaderelection resource RBAC: https://github.com/projectcontour/contour/pull/4204

Reply:
Yup I think that makes sense, the less permissions we need to have the better overall.